delta-contribs/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (2020) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.delta.storage._
+
+class COSLogStoreSuite extends LogStoreSuiteBase {
+
+  protected override def sparkConf = {
+    super.sparkConf.set(logStoreClassConfKey, logStoreClassName)
+      .set("spark.hadoop.fs.cos.atomic.write", "true")
+  }
+
+  override val logStoreClassName: String = classOf[COSLogStore].getName
+
+  testHadoopConf(
+    expectedErrMsg = ".*No FileSystem for scheme.*fake.*",
+    "fs.fake.impl" -> classOf[FakeFileSystem].getName,
+    "fs.fake.impl.disable.cache" -> "true")
+
+  protected def shouldUseRenameToWriteCheckpoint: Boolean = false
```

Comment:
this is not really testing on actual IBM cloud ... do you folks have full integration testing on your side to really test the atomic guarantees etc.?

Reply:
yeah. i dont have a good way. even our existing AzureLogStore and S3SingleDriverLogStore do not have cloud-specific unit tests for the same reasons... the complexity of maintaining it in a public repo. As long you folks have integration tests, that's good enough for now.